cluster-provision/k8s/1.17/manifests/cnao/operator.yaml

```diff
@@ -121,7 +121,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   annotations:
-    networkaddonsoperator.network.kubevirt.io/version: 0.32.0
+    networkaddonsoperator.network.kubevirt.io/version: 0.35.0
```

Comment:
Please change the commit and PR title+message so it reflects that we bump whole CNAO. You can mention that it is for macvtap in the message, but I don't think it is really needed. Bump is bump, latest greatest.

Reply:
I think you're trying to change the scope of this PR in a way that doesn't reflect what I want to do; I want to bump the version of a CNI/ dev plugin. 

CNAO is the means to an end, not the objective. That why I have the message written as is, and that's why I'm grabbing version 0.35.0 and not the latest. I don't care about anything on that container after 0.35.0 . While I said I can do it, since it's basically free, it's not my goal.

Reply:
It is as if you needed newer RPM, bumped OS version for it and commented about the RPM only, while drastically changing many of other dependencies.

If I browse the history to see why the heck did KubeMacPool version changed, I would look for notes about CNAO, not about macvtap-cni.

Reply:
That's because we apparently bump this by N versions at a time, thus updating N dependencies at once.

Could we start to update CNAO once it is updated, thus preventing these situations ?

I'll rephrase the commit message, since I have no alternative at all; but that doesn't mean I agree with this strategy.

Reply:
> That's because we apparently bump this by N versions at a time, thus updating N dependencies at once.

Single release of CNAO can carry several updated components or none, there is no set relation.

> 
> Could we start to update CNAO once it is updated, thus preventing these situations ?

Yes. Indeed one of many things we should automate.

Reply:
> > That's because we apparently bump this by N versions at a time, thus updating N dependencies at once.
> 
> Single release of CNAO can carry several updated components or none, there is no set relation.

I had assumed otherwise. What is the advantage of that if I may ask ? 

> 
> > Could we start to update CNAO once it is updated, thus preventing these situations ?
> 

Above I mean *bump* CNAO once it is updated; glad you understood what I meant anyway.
 
> Yes. Indeed one of many things we should automate.

Reply:
> > > That's because we apparently bump this by N versions at a time, thus updating N dependencies at once.
> > 
> > 
> > Single release of CNAO can carry several updated components or none, there is no set relation.
> 
> I had assumed otherwise. What is the advantage of that if I may ask ?

If we know we are going to bump multiple components during a single week, we can release only after it. Note that releasing in CNAO requires a dedicated PR with generated manifest bundles, it requires manual intervention and take a long time (again, needs automation).

CNAO can be seen as a distribution. We collect upstream components and when we think it makes sense (lack of defined rules) to release a new version with the set of components we have, we release.

Reply:
> > > > That's because we apparently bump this by N versions at a time, thus updating N dependencies at once.
> > > 
> > > 
> > > Single release of CNAO can carry several updated components or none, there is no set relation.
> > 
> > 
> > I had assumed otherwise. What is the advantage of that if I may ask ?
> 
> If we know we are going to bump multiple components during a single week, we can release only after it. Note that releasing in CNAO requires a dedicated PR with generated manifest bundles, it requires manual intervention and take a long time (again, needs automation).

I understand it's a cumbersome process, but that means it should be automated, not 'applied the least possible amount of times'. 

> 
> CNAO can be seen as a distribution. We collect upstream components and when we think it makes sense (lack of defined rules) to release a new version with the set of components we have, we release.

IMHO, the right thing to do is to bump CNAO for each individual upstream component change, as that is the 'unit' here. 

Anyway, I'll sleep on this for a couple of days, and will evaluate opening an issue where we can discuss the 'what' to automate, as this PR is far from the correct forum for this discussion; thanks for the answers.